### DISCUSSION TITLE
How to physically connect Raspberry to modbus.

### DISCUSSION THREAD

**Discussion by Takethis88** [veista/nilan#90] [Q&A] [answered]
Hello, I have a VP18cCom with a CTS 602 control panel. Software 2.35.a Version 1.04 (modbus 9 from what I've read here).
I am unsure about the hardware/electrical part of this project.  My Compact P has a USB 2 Type 2.0b cable accessible next to the water tank, can I use that to communicate with HA? If not, what is the best way to get access to the modbus? Do I need to remove the original CTS 602 control panel (and if yes what about the temperature sensor inside)? I've look at the existing discussions to see if these questions had already been answered but I couldn't find anything useful. I'm not familiar with RS485.
I'd like to keep everything working on the local network without having to rely on an external internet connection.

Thanks

**Comment by veista** [comment]
Hi. You need either a rs485 to Usb adapter or Modbus RTU to Modbus TCP bridge.

You can find the correct manuals from www.nilan.dk on how to hook everything up on your specific model. You can also ask Nilan support for instructions. Atleast in Finland they are very supportive and helpfull.

Compact P uses cts700 which is not yet supported or will be in the near future.

**Reply by Takethis88** [reply]
Compact P comes with both CTS 602 (old models) and then it's been switched to CTS 700 for newer models.
That's where I am now
![1](https://github.com/veista/nilan/assets/83803580/d110fa49-ce73-45f2-ae27-c55321001d35)
![2](https://github.com/veista/nilan/assets/83803580/2d4d6e01-aaae-4877-b894-273c63804cb9)
![3](https://github.com/veista/nilan/assets/83803580/a9682fb8-e6ee-4913-9dc7-706ccf18f4d7)

Unfortunately where I live Nilan getting through someone technical at Nilan has been an issue, I've been trying for 3 weeks to have a quote to update the old CTS 602 button interface to the touch display with no luck.
![image](https://github.com/veista/nilan/assets/83803580/bee3139b-6c10-4102-8e47-91161fbf94a9)
From the image above I think the RS485 port is at the bottom left of the logic board.

**Reply by veista** [reply]
You should make your connections to the 8pole connector.

**Reply by Takethis88** [reply]
Thanks, I'll check that. To be honest I can't see it. It might be behind the CTS interface. From this https://www.en.nilan.dk/Files/Files/Engelsk/Archive/Software%20instructions/2016-02%20CTS602_Modbus_protokol.pdf I thought I could connect to pin 2,3 and 6. 
![image](https://github.com/veista/nilan/assets/83803580/613d40b2-26f1-45cb-b8ce-80abb9876ea5)

**Reply by veista** [reply, marked as answer]
I believe that is correct.
